### PR TITLE
Android: Animate covers on focus change

### PR DESCRIPTION
**Pull request by t895**
Subtle animation when using controller navigation.
animate-cover

### CODE REVIEW

**Review by JosJuice — APPROVED** [review]

**Comment by K0bin** [comment]
It would probably be better to cache the animations and not reload them every time.

**Comment by t895** [comment]
> It would probably be better to cache the animations and not reload them every time.

Got it

**Review by K0bin — APPROVED** [review]

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.java [hunk 1]
```diff
@@ -83,6 +85,22 @@ public void onBindViewHolder(GameViewHolder holder, int position)
...
+    holder.binding.getRoot().setOnFocusChangeListener((v, hasFocus) ->
+    {
+      if (hasFocus)
+      {
+        holder.binding.cardGameArt.startAnimation(animateIn);
+      }
+      else
+      {
+        holder.binding.cardGameArt.startAnimation(animateOut);
+      }
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.java:102]: If you want to make it look a little neater:

```suggestion
      holder.binding.cardGameArt.startAnimation(hasFocus ? animateIn : animateOut);
```

Comment (t895) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.java]: I like that

**Comment by MayImilae** [comment]
The grey colour in the demo isn't amazing, it's a little dark so black text on mid grey background is pretty eh, but that's an easy detail to fix later. The animation itself is good. LGTM.

**Comment by t895** [comment]
> The grey colour in the demo isn't amazing, it's a little dark so black text on mid grey background is pretty eh, but that's an easy detail to fix later. The animation itself is good. LGTM. 

Yeah that's just the default "focus" color in Android. I'll find something better later.

**Pull request merged**